protostar-rust/src/lib.rs

```diff
@@ -137,6 +174,35 @@ fn run_tests(tests: TestsFromFile, tests_stats: &mut TestsStats) -> Result<()> {
     Ok(())
 }
 
+fn filter_tests_by_name(
+    test_name_filter: Option<&str>,
+    exact_match: bool,
+    test_configs: Vec<TestConfig>,
+) -> Result<Vec<TestConfig>> {
+    let mut result = vec![];
+    if let Some(test_name_filter) = test_name_filter {
+        for test in test_configs {
+            if exact_match {
+                if test.name == test_name_filter {
+                    result.push(test);
+                }
+            } else {
+                let name = test
+                    .name
+                    .rsplit("::")
+                    .next()
+                    .context(format!("Failed to get test name from = {}", test.name))?;
+                if name.contains(test_name_filter) {
+                    result.push(test);
+                }
+            }
+        }
+    } else {
+        result = test_configs;
+    }
```

Comment:
Filter would make it more complicated. We're doing error handling in the `rsplit` and afaik you cannot easily bubble errors from closures. 

It can be done but would result it more complicated code.

Reply:
Bubbling errors is a good point tbh

Reply:
Discussed online, my misunderstanding of the code

Reply:
I've tried and this cannot be simply replaced with filter_map, there's still a problem of propagating errors from it.